File: Phase_3/frontend/src/app/services/report.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { MonthlyAdoptionReport } from '../models/monthly-adoption-report.model';
import { ACMainReport } from '../models/ac-main-report.model';
import { ACDrillAdopt } from '../models/ac-drill-adopt.model';
import { ACDrillExpense } from '../models/ac-drill-expense.model';
import { ACDrillSurrender } from '../models/ac-drill-surrender.model';
import { Volunteer } from '../models/volunteer';
import { ExpenseAnalysisReport } from '../models/expense-analysis-report.model';
import { AdoptionReviewReport } from '../models/adoption-review.report.model';

@Injectable({
  providedIn: 'root'
})
export class ReportService {

 private baseUrl = 'http://localhost:8080/report';
 
  constructor(private http: HttpClient) { }
 
   getMonthlyAdoptionReport(): Observable<MonthlyAdoptionReport[]> {
     return this.http.get<MonthlyAdoptionReport[]>(`${this.baseUrl}/monthly-adoption`);
   }

   getExpenseAnalysisReport(): Observable<ExpenseAnalysisReport[]> {
    return this.http.get<ExpenseAnalysisReport[]>(`${this.baseUrl}/expense-analysis`);
  }

   getACMainReport(): Observable<ACMainReport[]> {
    return this.http.get<ACMainReport[]>(`${this.baseUrl}/ac-main-report`);
  }

  getACDrillAdopt(monthYear: any): Observable<ACDrillAdopt[]> {
    return this.http.get<ACDrillAdopt[]>(`${this.baseUrl}/ac-drill-adopt/${monthYear}`);
  }

  getACDrillExpense(monthYear: any): Observable<ACDrillExpense[]> {
    return this.http.get<ACDrillExpense[]>(`${this.baseUrl}/ac-drill-expense/${monthYear}`);
  }

  getACDrillSurrender(monthYear: any): Observable<ACDrillSurrender[]> {
    return this.http.get<ACDrillSurrender[]>(`${this.baseUrl}/ac-drill-surrender/${monthYear}`);
  }

  getVolunteerLookup(name: any): Observable<Volunteer[]> {
    return this.http.get<Volunteer[]>(`${this.baseUrl}/volunteer/${name}`);
  }

  getVolunteerBirthdayReport(month?: string, year?: string): Observable<Volunteer[]> {
    return this.http.get<Volunteer[]>(`${this.baseUrl}/volunteer-birthday/${month}/${year}`);
  }

  getAdoptionReview(): Observable<AdoptionReviewReport[]>{
    return this.http.get<AdoptionReviewReport[]>(`${this.baseUrl}/adoption-review`);
  }

}

File: Phase_3/backend/burdells-devoted-dogs/src/main/java/edu/gatech/cs/dogrescue/services/ReportService.java
package edu.gatech.cs.dogrescue.services;

import java.sql.SQLException;
import java.util.List;

import edu.gatech.cs.dogrescue.dao.ReportDAO;
import edu.gatech.cs.dogrescue.dto.*;
import jakarta.enterprise.context.RequestScoped;
import jakarta.inject.Inject;

@RequestScoped
public class ReportService {
    @Inject
    ReportDAO reportDAO;

    public List<MonthlyAdoptionReport> getMonthlyAdoptionReport() throws SQLException {
        return reportDAO.getMonthlyAdoptionReport();
    }

    public List<ExpenseAnalysisReport> getExpenseAnalysisReport() throws SQLException {
        return reportDAO.getExpenseAnalysisReport();
    }

    public List<ACMainReport> getACMainReport() throws SQLException {
        return reportDAO.getACMainReport();
    }

    public List<ACDrillSurrender> getACDrillSurrender(String MonthYear) throws SQLException {
        return reportDAO.getACDrillSurrender(MonthYear);
    }

    public List<ACDrillAdopt> getACDrillAdopt(String MonthYear) throws SQLException {
        return reportDAO.getACDrillAdopt(MonthYear);
    }

    public List<ACDrillExpense> getACDrillExpense(String MonthYear) throws SQLException {
        return reportDAO.getACDrillExpense(MonthYear);
    }

    public List<User> getVolunteerLookup(String name) throws SQLException {
        return reportDAO.getVolunteerLookup(name);
    }

    public List<User> getVolunteerLookup() throws SQLException {
        return reportDAO.getVolunteerLookup();
    }

    public List<Volunteer> getVolunteerBirthDayReport(String month, String year) throws SQLException {
        return reportDAO.getVolunteerBirthday(month, year);
    }

    public List<AdoptionReview> getAdoptionReviewReport() throws SQLException {
        return reportDAO.getAdoptionReviewReport();
    }
}
